platform-hub-api/app/controllers/users_controller.rb

```diff
@@ -49,6 +58,42 @@ def revoke_admin
     head :no_content
   end
 
+  # POST /users/:id/activate
+  def activate
+    keycloak_agent_service.activate_user(@user)
+
+    @user.activate!
+
+    AuditService.log(
+      context: audit_context,
+      action: 'make_active',
+      auditable: @user
+    )
+
+    head :no_content
+  rescue => e
+    render_error "User activation failed - #{e.class}: #{e.message}",
+                :unprocessable_entity and return
+  end
+
+  # POST /users/:id/deactivate
+  def deactivate
+    keycloak_agent_service.deactivate_user(@user)
+
+    @user.deactivate!
```

Comment:
Is it worth moving this logic into a separate service (e.g. `UserActivationService`) so that the two steps of Keycloak then local hub activation/deactivation are encapsulated in one place (and thus could be used through the console, or background jobs)?